### ISSUE TITLE
Ways to automatically discover Deno ports

### ISSUE THREAD

**Issue by KillyMXI** [garronej/denoify#35]
It's not something that affects me directly yet, but I find it a curious and underexplored problem.

The situation may get exponentially better as more packages expose their Deno port in some way suitable for automation. Much better than trying to keep a hand-filled dictionary of ports.

I was considering `make-deno-edition` along with Denoify, and their attempts to address this problem is what made me aware and curious about this problem. Although their solutions don't fill me with confidence:

- "deno" field in `package.json` pointing to an entry module takes the very basic and lucrative name and uses it for something very limited. I have a package with multiple exports and I wouldn't be able to describe it properly. Extended spec would be needed.

    Somehow I feel extending the "exports" section to accept "deno" key besides "import", "require", etc. would make much more sense. Although that may have compatibility issues.

    Much simpler use for "deno" field (if ever needed) would be just to serve in the same way as "denoify.out" does - to point at the folder where Deno-targeted code resides, without specifying an entry point (that can potentially be recovered from "exports" section). Or it could be an url to the port.

- "editions" section seems descriptive for humans but not very actionable for automation. Extra layer of conventions would require on top of what already feels taking much more cognitive load compared to what it offers. (Maybe I just don't work with frontend to fully appreciate it though.)

Things that I think Denoify could do on it's own with `package.json` of dependency packages:

- Check the presence of "denoify" section and "repository" section;
- If <https://deno.land/x> can be searched by a repository name then you've got the port;
- Or <https://deno.land/x> can be checked for the package name and then validated by the repository name to make sure it's the same package;
- If "denoify.out" is available then it might be possible to check the repo for the presence of that folder and make use of it.

**Comment by garronej** [comment]
> But I think it also exposes the issue with documentation.  

Yeah, that's true 😞 

> Understanding the algorithm would be more useful.

The problem is that:  
1) It is very sophisticated, it's description would have to be a long text that I know very few people would read, if any. 
2) If I update the algorithm I have to update the documentation accordingly. Just keeping [the demo repo](https://github.com/garronej/my_dummy_npm_and_deno_module) up to date is easier. 

> Although I'm not sold on some practices. In particular, I wouldn't accept a PR with "like 2 lines" as linked above. 

I didn't see it under this angle. I now understand better why it wasn't merged. 

There is always the possibility of using a custom [a custom .deno.ts](https://github.com/garronej/my_dummy_npm_and_deno_module#step-1-dealing-with-unsupported-node-builtins) module for things that can't easily be ported... 

How sucessfull where you denoifying node-html-to-text so far?

**Comment by garronej** [comment]
Hi @KillyMXI,  
I finally dealt with this issue.  
Now, if you upgrade denoify to v0.15 and release a new version of `leac` and `peberminta` they will be automaticallty discoverable.  

Meaning that this import:  
```typescript
import { createLexer } from "leac";
```
Will be tranpiled into:  
```typescript
import { createLexer } from "https://raw.githubusercontent.com/mxxii/leac/v0.5.2/deno/mod.ts";
```

I created a documentation page: https://docs.denoify.land/automatic-discovery-of-ports

**Comment by KillyMXI** [comment]
Thank you.

I hope this doesn't break the discoverability of the second entry point of `peberminta`.
At least it keeps working with new `denoify` version and existing explicit `ports`.
Will check again when I publish new versions.

**Comment by garronej** [comment]
It shouldn't break anything. If it dose please report. Best regards

**Comment by KillyMXI** [comment]
I found a difference:
If I still have ports specified in my `package.json`, `denoify` uses it's own resolution algorithm anyway.
It would make sense to treat my input with higher priority and let me override paths.

**Comment by garronej** [comment]
You are right, it's fixed in 1.0.1
